### PR TITLE
Add 1-WPK day 5

### PR DESCRIPTION
**Pull request by camargodev**

### CODE REVIEW

File: hackerrank/1-week-prep/day-5/balanced-brackets.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        
+    def is_empty(self):
+        return self._size == 0
+        
+    def push(self, value):
+        self._size += 1
+        self._values.append(value)
+    
+    def top(self):
+        return self._values[self._size-1]
```
Comment (jdalbosco) [inline hackerrank/1-week-prep/day-5/balanced-brackets.py]:
```suggestion
        return self._values[-1]
```

File: hackerrank/1-week-prep/day-5/merge-two-sorted-linked-lists.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+def mergeLists(head1, head2):
```
Comment (jdalbosco) [inline hackerrank/1-week-prep/day-5/merge-two-sorted-linked-lists.py]:
```suggestion
def merge_lists(head1, head2):
```

File: hackerrank/1-week-prep/day-5/merge-two-sorted-linked-lists.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        else:
+            if merged_list_head is None:
+                merged_list_head = head2
+            current, last, head2 = consume_list(current, last, head2)
+    if head1 is None:
+        while head2 is not None:
+            current, last, head2 = consume_list(current, last, head2)
+    if head2 is None:
+        while head1 is not None:
+            current, last, head1 = consume_list(current, last, head1)
```
Comment (jdalbosco) [inline hackerrank/1-week-prep/day-5/merge-two-sorted-linked-lists.py]:
```suggestion
    if head1 is None:
        current.next = head2
    if head2 is None:
        current.next = head1
```

File: hackerrank/1-week-prep/day-5/mock-test-5.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+def pairs(exp_difference, values):
+    num_pairs = 0
+    occurrences = dict()
+
+    for value in values:
+        if value not in occurrences:
+            occurrences[value] = 0
+        occurrences[value] += 1
```
Comment (jdalbosco) [inline hackerrank/1-week-prep/day-5/mock-test-5.py:8]: One of the constraints of the problem is that each `value` in `values` will be unique, so you didn't have to keep track of occurrences.

File: hackerrank/1-week-prep/day-5/mock-test-5.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    for value in values:
+        pair_1 = value - exp_difference
+        pair_2 = value + exp_difference
+
+        if pair_1 in occurrences:
+            num_pairs += occurrences[pair_1]
+        if pair_2 in occurrences:
+            num_pairs += occurrences[pair_2]
+
+    return num_pairs//2
```
Comment (jdalbosco) [inline hackerrank/1-week-prep/day-5/mock-test-5.py:19]: You can just validate each of the values with the previous - or the following - ones. No need to validate with both and then divide by two. 😸

File: hackerrank/1-week-prep/day-5/queue-using-two-stacks.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        self._size += 1
+        self._values.append(value)
+    
+    def top(self):
+        return self._values[self._size-1]
+    
+    def pop(self):
+        value = self._values[self._size-1]
+        self._size -= 1
+        self._values.pop()
```
Comment (jdalbosco) [inline hackerrank/1-week-prep/day-5/queue-using-two-stacks.py]:
```suggestion
        self._size -= 1
        value = self._values.pop()
```

File: hackerrank/1-week-prep/day-5/queue-using-two-stacks.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    def dequeue(self):
+        if self._stack_1.is_empty():
+            while not self._stack_2.is_empty():
+                self._stack_1.push(self._stack_2.pop())
+        self._stack_1.pop()
+    
+    def print_front(self):
+        if self._stack_1.is_empty():
+            while not self._stack_2.is_empty():
+                self._stack_1.push(self._stack_2.pop())
```
Comment (jdalbosco) [inline hackerrank/1-week-prep/day-5/queue-using-two-stacks.py]: Lines 34-36 and 40-42 are the same, maybe you could create a function for those operations.

File: hackerrank/1-week-prep/day-5/queue-using-two-stacks.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    def pop(self):
+        value = self._values[self._size-1]
+        self._size -= 1
+        self._values.pop()
+        return value
+        
+class Queue:
+    def __init__(self):
+        self._stack_1 = Stack()
+        self._stack_2 = Stack()
```
Comment (jdalbosco) [inline hackerrank/1-week-prep/day-5/queue-using-two-stacks.py]: Just a suggestion, but I think using more informative names for these stacks will improve readability.

**Review by jdalbosco — CHANGES_REQUESTED** [review]

File: hackerrank/1-week-prep/day-5/queue-using-two-stacks.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    def dequeue(self):
+        if self._stack_1.is_empty():
+            while not self._stack_2.is_empty():
+                self._stack_1.push(self._stack_2.pop())
+        self._stack_1.pop()
+    
+    def print_front(self):
+        if self._stack_1.is_empty():
+            while not self._stack_2.is_empty():
+                self._stack_1.push(self._stack_2.pop())
```
Comment (camargodev) [inline hackerrank/1-week-prep/day-5/queue-using-two-stacks.py]:
```suggestion
class Queue:
    def __init__(self):
        self._main_stack = Stack()
        self._helper_stack = Stack()
    
    def enqueue(self, new_value):
        if self._main_stack.is_empty() and self._helper_stack.is_empty():
            self._main_stack.push(new_value)
        else:
            self._helper_stack.push(new_value)
    
    def dequeue(self):
        self._move_to_main_stack()
        self._main_stack.pop()
    
    def print_front(self):
        self._move_to_main_stack()
        print(self._main_stack.top())
        
    def _move_to_main_stack(self):
        if not self._main_stack.is_empty():
            return
        while not self._helper_stack.is_empty():
            self._main_stack.push(self._helper_stack.pop())
```

File: hackerrank/1-week-prep/day-5/queue-using-two-stacks.py [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    def print_front(self):
+        self._move_to_main_stack()
+        print(self._main_stack.top())
+        
+    def _move_to_main_stack(self):
+        if not self._main_stack.is_empty():
+            return
+        while not self._helper_stack.is_empty():
+            self._main_stack.push(self._helper_stack.pop())
+        print(self._stack_1.top())
```
Comment (jdalbosco) [inline hackerrank/1-week-prep/day-5/queue-using-two-stacks.py]:
```suggestion
```

**Review by jdalbosco — APPROVED** [review]

**Pull request merged**